homeassistant/components/nobo_hub/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""The Nobø Ecohub integration."""
+from __future__ import annotations
+
+import logging
+
+from pynobo import nobo
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_SERIAL, DOMAIN
+
+PLATFORMS = ["climate"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Nobø Ecohub from a config entry."""
+
+    serial = entry.data.get(CONF_SERIAL)
+    ip_address = entry.data.get(CONF_IP_ADDRESS)
+    discover = ip_address is None
+    hub = nobo(serial=serial, ip=ip_address, discover=discover, loop=hass.loop)
```

Comment:
This is to pass the event loop to the `pynobo` library, otherwise it calls `asyncio.get_event_loop()` and fails:
```
  File "/Users/omw/proj/home-assistant/homeassistant/components/nobo_hub/__init__.py", line 25, in async_setup_entry
    hub = nobo(serial=serial, ip=ip_address, discover=discover)
  File "/Users/omw/proj/home-assistant/venv/lib/python3.9/site-packages/pynobo.py", line 227, in __init__
    self._loop.run_until_complete(self.start())
  File "/Users/omw/.pyenv/versions/3.9.9/lib/python3.9/asyncio/base_events.py", line 618, in run_until_complete
    self._check_running()
  File "/Users/omw/.pyenv/versions/3.9.9/lib/python3.9/asyncio/base_events.py", line 578, in _check_running
    raise RuntimeError('This event loop is already running')
RuntimeError: This event loop is already running
```

Reply:
The library needs updating, as this will break Python 3.10

Reply:
I see. I'll look into it, and follow up here as soon as a new version is released.

Reply:
This still needs addressing.